include/url.h

```diff
@@ -0,0 +1,86 @@
+#ifndef URL_CPP_H
+#define URL_CPP_H
+
+#include <stdexcept>
+#include <string>
+#include <vector>
+#include <unordered_map>
+#include <unordered_set>
+
+namespace Url {
+
+    struct UrlParseException : public std::logic_error {
+        UrlParseException(const std::string& message): std::logic_error(message) {}
```

Comment:
Space before colon for the initializer list if you want to follow Martin's new formatting criteria.